http3/headers.go

```diff
@@ -0,0 +1,264 @@
+package http3
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/lucas-clemente/quic-go/quicvarint"
+	"github.com/marten-seemann/qpack"
+	"golang.org/x/net/http/httpguts"
+	"golang.org/x/net/idna"
+)
+
+const (
+	pseudoHeaderMethod    = ":method"
+	pseudoHeaderScheme    = ":scheme"
+	pseudoHeaderAuthority = ":authority"
+	pseudoHeaderPath      = ":path"
+	pseudoHeaderProtocol  = ":protocol"
+	pseudoHeaderStatus    = ":status"
+)
+
+// RequestHeaders returns valid HTTP/3 header fields for req, or an error if req
+// is malformed.
+func RequestHeaders(req *http.Request) ([]qpack.HeaderField, error) {
```

Comment:
Not sure where this is coming from. I assume that you renamed a file and now Git recognizes it as new? It would be helpful if you revert the renaming, so Git can show the actual changes that happened here.

Reply:
It wasn’t so much of a rename as a refactor of `(*requestWriter).writeHeaders`. The rest of `requestWriter` was subsumed by `http3.client` and `http3.RequestStream`.